packages/gotrue/lib/src/gotrue_client.dart

```diff
@@ -479,8 +479,69 @@ class GoTrueClient {
     return refreshCompleter.future;
   }
 
+  /// Sends a reauthentication OTP to the user's email or phone number.
+  ///
+  /// Requires the user to be signed-in.
+  Future<void> reauthenticate() async {
+    final session = currentSession;
+    if (session == null) {
+      throw AuthException('Not logged in.');
+    }
+
+    final options =
+        GotrueRequestOptions(headers: headers, jwt: session.accessToken);
+
+    await _fetch.request(
+      '$_url/reauthenticate',
+      RequestMethodType.get,
+      options: options,
+    );
+  }
+
+  ///Resends an existing signup confirmation email, email change email, SMS OTP or phone change OTP.
+  ///
+  ///Use [EmailResendParams] or [PhoneResendParams] to specify the type of resend.
+  Future<ResendResponse> resend({
+    String? email,
+    String? phone,
+    required OtpType type,
+    String? captchaToken,
+  }) async {
+    assert((email != null && phone == null) || (email == null && phone != null),
+        '`email` or `phone` needs to be specified.');
+
+    _removeSession();
```

Comment:
I've copied that from gotrue-js, but I think always removing the session is wrong, isn't it? If the use just wants to update their email and has to resend that, we don't want to remove the session?

Best way is probably like in `verifyOTP`,right?
```suggestion
    if (type != OtpType.emailChange && type != OtpType.phoneChange) {
      _removeSession();
    }
```

Reply:
Yeah, I think you are correct! Maybe you could also comment in the gotrue PR to let them know about this one?